src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -16,647 +16,100 @@
  */
 
 #include <app/clusters/scenes-server/SceneTableImpl.h>
+#include <app/common/FabricTableImpl.hpp>
 #include <lib/support/DefaultStorageKeyAllocator.h>
 #include <stdlib.h>
 
-namespace chip {
-namespace scenes {
+using namespace chip;
+using namespace chip::scenes;
+using namespace chip::app::common;
 
+namespace {
 /// @brief Tags Used to serialize Scenes so they can be stored in flash memory.
-/// kSceneCount: Number of scenes in a Fabric
-/// kStorageIDArray: Array of StorageID struct
-/// kEndpointID: Tag for the Endpoint ID to which this scene applies to
 /// kGroupID: Tag for GroupID if the Scene is a Group Scene
 /// kSceneID: Tag for the scene ID together with the two previous tag, forms the SceneStorageID
 /// kName: Tag for the name of the scene
 /// kTransitionTime: Tag for the transition time of the scene in miliseconds
 enum class TagScene : uint8_t
 {
-    kEndpointSceneCount = 1,
-    kSceneCount,
-    kStorageIDArray,
-    kGroupID,
+    kGroupID = 4,
```

Comment:
This seems to be assuming something about something.  Where is this 4 coming from?

Reply:
Updated to reference, apologies